### PR TITLE
Add Bip32

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by MementoRC** [comment]
@conda-forge/help-python-c, hello, ready for review

File: recipes/bip32/build.sh
Comment (carterbox) [inline recipes/bip32/build.sh:5]:
```suggestion
$PYTHON -m pip install . --no-build-isolation \
    --no-deps --ignore-installed --no-index --no-cache-dir -vv --use-pep517
```
If you call pip install from a script instead of from in the recipe (as in the example recipe) you need to add all the args that are added automatically.

File: recipes/bip32/bld.bat
Comment (carterbox) [inline recipes/bip32/bld.bat:2]: It should be able to handle that syntax.

File: recipes/coincurve/meta.yaml
Comment (carterbox) [inline recipes/coincurve/meta.yaml:1]: This is a duplicate recipe. Coincurve has it's own PR. Please combine all of your related PRs into one or wait until the coincurve PR is merged.

**Review by carterbox — CHANGES_REQUESTED** [review]

**Comment by MementoRC** [comment]
Understood, I just wanted to go through the whole build, since I submitted
all these, I intended to revise them as they got accepted. This was
suggested in the documentation.

Many of the submissions I made follow this framework,  in particular the
injective-py, perhaps looking at it last would be beneficial

Thank you, it feels like I was gifted a wish!

File: recipes/coincurve/meta.yaml
Comment (MementoRC) [reply recipes/coincurve/meta.yaml:1]: I need to re-think how I organize the recipes I submitted. I do want to test them through the flow (in particular since windows is so foreign to me).

Is the best way to move this recipe (and the other similar ones) back to draft?

How to remove the `review-requested` flag to avoid reviewers going through the recipes that are moved back to draft?

From your experience, and given that all the recipes are passing checks, what would be the more productive strategy?
   - Focus on individual packages and having to wait through all the individual PRs to be merged
   - Focus on the final packages that I need (with the 5+ individual packages bundled)?
   
Thank you

Comment (carterbox) [reply recipes/coincurve/meta.yaml:1]: I think I have to remove the labels for you because you aren't a member of the conda-forge org yet. Yes, reverting them to a draft is a good idea. You can also debug your recipes on your local machine (if you have a Windows machine) because you can run conda-build from the command line. Just use the conda-forge channel for all of the packages and the `--use-local` flag.

You should consider the dependency tree of the ecosystem that you are trying to build and break it into PRs by theme/dependency For example, all the binding packages in one PR, coinbase and the libsec256k1 in another PR etc.

Comment (MementoRC) [reply recipes/coincurve/meta.yaml:1]: I have used some of the `build-locally.py`, but it is not foolproof (for example for locally installed packages) and I do not have a mac or windows (which is really a tremendous pain for me, likely because I know very little about it)

If you don't mind, can we start with `libsecp256k1`? (https://github.com/conda-forge/staged-recipes/pull/23882). This is the one I worked on the most

The `coincurve_dynamic` is then the second one that I think simplifies and implements a few of the feedback you gave me so far

I don't think it is worth our time to work on `coincurve` since `coincurve_dynamic` is much simpler, I would rather close it

**Comment by MementoRC** [comment]
This recipe is awaiting merge of `coincurve`, upon which the local `coincurve` recipe provided to pass the checks will be remove

**Comment by MementoRC** [comment]
@carterbox This should be ready for review now that coincurve is on conda-forge

**Comment by carterbox** [comment]
I'm seeing the tests failing, please fix that before someone comes along to review. I may or may not be reviewing this weekend.

**Comment by MementoRC** [comment]
@carterbox Right, the version of coincurve we worked on is 19.0.0, but the recipe calls for 18.0.0, I just updated it to back-populate 18.0.0 (although, I can only do so by using PYPI package as we discussed)
Close/Open to restart the checks

File: recipes/bip32/build.sh
Comment (MementoRC) [reply recipes/bip32/build.sh]: @carterbox I only understood this yesterday, while working on the PR for `oincurve-feedstock`!

**Comment by MementoRC** [comment]
@carterbox This recipe is ready for review. Note that I used `coincurve 18.0.0`. The upstream maintainer does not seem active

**Comment by MementoRC** [comment]
@carterbox I have updated `coincurve` requirement to 19.0.0 - The author is intending to release a new version once we address his static-built-coincurve-windows issue

**Comment by MementoRC** [comment]
@carterbox I added a patch of the requirements to make the check pass (or highlight if other issues arise)

**Pull request merged**
